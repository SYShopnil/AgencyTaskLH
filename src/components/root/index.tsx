import Button from './button';

export { Button };
export { BannerSectionBar } from './banner-bar';
export { SlideCard } from './slide-card/';
export { HeroSectionWithOnScrollRotation } from './hero-section-with-on-scroll-rotaition';
export { Navbar } from './navbar';
