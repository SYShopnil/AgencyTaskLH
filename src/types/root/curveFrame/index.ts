import { ReactNode } from 'react';

export enum CurveFrameVariantEnum {
  with_out_gradient = 'with_out_gradient',
  with_gradient = 'with_gradient',
}

export interface ICurveFrame {
  children: ReactNode;
  variant?: CurveFrameVariantEnum;
  dataTestIdForCurveFrame?: string;
  borderWidth: string;
}
