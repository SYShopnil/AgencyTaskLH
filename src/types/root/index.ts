export type { IButton, INormalButton, IRedirectButton } from './Button/';

export { BtnColorSchema } from './Button/';

export type { IBtnCard } from './btn-card';
export { ColorSchema, CardSize } from './btn-card';

export type { IIconStore } from './_icon';
export { IconName as IconEnums } from './_icon';

export type { ICurveFrame } from './curveFrame';
export { CurveFrameVariantEnum } from './curveFrame';

export type { IInfoCard } from './infoCard';
export type { ITitleDescSection } from './title_desc_sec/';

export type { IBanner, IBannerSectionBar } from './banner-bar/banner.ts';
export type { ISlideCard } from './slide-card';

export type { IHeroSectionWithOnScrollRotation } from './hero-section-with-on-scroll-rotaition/index';
export type { IFooter } from './footer';

export type {
  ICommonSection,
  IMegaMenu,
  IMegaMenuItems,
  ISubMenuBlock,
  INavbar,
} from './navbar';
