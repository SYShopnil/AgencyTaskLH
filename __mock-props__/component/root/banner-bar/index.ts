import { IBanner, IBannerSectionBar } from '@src/types/root';
import { IconName } from '@src/types/root/_icon';

export const BannerImageMockData: IBanner[] = [
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
  {
    iconName: IconName.Treva,
    clickableLink: 'https://www.youtube.com/',
  },
];

export const BannerBarSectionMockData: IBannerSectionBar = {
  contents: BannerImageMockData,
  bannerBarSectionTestId: 'mainWrapper',
};
