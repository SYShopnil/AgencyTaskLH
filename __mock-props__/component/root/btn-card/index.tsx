import { CardSize, ColorSchema, IBtnCard } from '@src/types/root';

export const btnCardMockProps: IBtnCard[] = [
  {
    colorSchema: ColorSchema.PrimaryBlueBgTextWhite,
    title: 'Grass',
    cardSize: CardSize.card_lg,
    fill: 65,
    data_test_id: 'btnCardOne',
  },
  {
    colorSchema: ColorSchema.PrimaryBlueBgTextWhite,
    title: 'Grass',
    cardSize: CardSize.card_sm,
    fill: 50,
    data_test_id: 'btnCardTwo',
  },
];
