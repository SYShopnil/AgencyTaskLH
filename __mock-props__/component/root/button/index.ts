import { BtnColorSchema, IRedirectButton } from '@src/types/root';

export const mock_data_redirectButton: IRedirectButton[] = [
  {
    btnText: 'Back to Homepage',
    colorSchema: BtnColorSchema.BlackBgTextWhite,
    isOpenNewTab: true,
    redirectLink: 'https://chat.openai.com/',
    dataTestId: 'Redirect Button One',
  },
  {
    btnText: 'Back to Homepage Two',
    colorSchema: BtnColorSchema.BlackBgTextWhite,
    isOpenNewTab: false,
    redirectLink: '',
    dataTestId: 'Redirect Button Two',
  },
];
