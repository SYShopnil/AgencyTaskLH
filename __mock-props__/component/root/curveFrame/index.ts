import { CurveFrameVariantEnum, ICurveFrame } from '@src/types/root';

export const mockPropsOfCurveFrame: ICurveFrame[] = [
  {
    variant: CurveFrameVariantEnum.with_gradient,
    children: `<div}>
                <p>Hello world</p>
                <p>Hello world</p>
                <p>Hello world</p>
                <p>Hello world</p>
            </div>`,
    dataTestIdForCurveFrame: 'frameCurveOne',
    borderWidth: '2%',
  },
  {
    variant: CurveFrameVariantEnum.with_out_gradient,
    children: `<div}>
                <p>Hello world</p>
                <p>Hello world</p>
                <p>Hello world</p>
                <p>Hello world</p>
            </div>`,
    dataTestIdForCurveFrame: 'frameCurveTwo',
    borderWidth: '2%',
  },
];
