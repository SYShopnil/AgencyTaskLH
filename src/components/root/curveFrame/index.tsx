import { CurveFrameVariantEnum, ICurveFrame } from '@src/types/root';
import React from 'react';
import styles from './index.module.scss';

const CurveFrame = ({
  children,
  variant,
  dataTestIdForCurveFrame,
  borderWidth,
}: ICurveFrame) => {
  let setBgBasedOnVariant = '';
  switch (variant) {
    case CurveFrameVariantEnum.with_gradient: {
      setBgBasedOnVariant = `bg-primaryGradient`;
      break;
    }
    case CurveFrameVariantEnum.with_out_gradient: {
      setBgBasedOnVariant = `bg-[#FFFFFF]`;
      break;
    }
    default: {
      setBgBasedOnVariant = `bg-primaryGradient`;
    }
  }
  return (
    <div
      className={`${styles.filter_container} `}
      data-testid={dataTestIdForCurveFrame}
    >
      <div
        className={`${styles.box_Outer} ${setBgBasedOnVariant}`}
        style={{ padding: borderWidth }}
      >
        <div className={`${styles.filter_container}`}>
          <div className={`${styles['box_Inner']} bg-white`}>
            <span>{children}</span>
          </div>
        </div>
      </div>
      <svg
        style={{ visibility: 'hidden', position: 'absolute' }}
        width="0"
        height="0"
        version="1.1"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <filter id="goo">
            <feGaussianBlur in="SourceGraphic" stdDeviation="3" result="blur" />
            <feColorMatrix
              in="blur"
              mode="matrix"
              values="1 0 0 0 0  0 1 0 0 0  0 0 1 0 0  0 0 0 19 -9"
              result="goo"
            />
            <feComposite in="SourceGraphic" in2="goo" operator="atop" />
          </filter>
        </defs>
      </svg>
    </div>
  );
};

export default CurveFrame;
