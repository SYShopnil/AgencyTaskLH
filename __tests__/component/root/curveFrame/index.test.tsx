import { mockPropsOfCurveFrame } from '@root/__mock-props__/component/root/curveFrame';
import CurveFrame from '@src/components/root/curveFrame';
import { render } from '@testing-library/react';

let curveTracker = 0;

describe.each(mockPropsOfCurveFrame)(
  `When curve frame has rendered`,
  (curveFrame) => {
    it(`Expect component ${++curveTracker} has rendered successfully`, () => {
      const { getByTestId } = render(
        <CurveFrame {...curveFrame}>{curveFrame.children}</CurveFrame>
      );
      const getCurveFrame = getByTestId(
        curveFrame.dataTestIdForCurveFrame || ''
      );
      expect(getCurveFrame).toBeInTheDocument();
      expect(getCurveFrame).toMatchSnapshot();
    });
  }
);
