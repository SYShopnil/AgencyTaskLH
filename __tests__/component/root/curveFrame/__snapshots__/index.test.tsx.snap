// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When curve frame has rendered Expect component 1 has rendered successfully 1`] = `
<div
  class="filter_container "
  data-testid="frameCurveOne"
>
  <div
    class="box_Outer bg-primaryGradient"
    style="padding: 2%;"
  >
    <div
      class="filter_container"
    >
      <div
        class="box_Inner bg-white"
      >
        <span>
          &lt;div}&gt;
                &lt;p&gt;Hello world&lt;/p&gt;
                &lt;p&gt;Hello world&lt;/p&gt;
                &lt;p&gt;Hello world&lt;/p&gt;
                &lt;p&gt;Hello world&lt;/p&gt;
            &lt;/div&gt;
        </span>
      </div>
    </div>
  </div>
  <svg
    height="0"
    style="visibility: hidden; position: absolute;"
    version="1.1"
    width="0"
    xmlns="http://www.w3.org/2000/svg"
  >
    <defs>
      <filter
        id="goo"
      >
        <fegaussianblur
          in="SourceGraphic"
          result="blur"
          stdDeviation="3"
        />
        <fecolormatrix
          in="blur"
          mode="matrix"
          result="goo"
          values="1 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 19 -9"
        />
        <fecomposite
          in="SourceGraphic"
          in2="goo"
          operator="atop"
        />
      </filter>
    </defs>
  </svg>
</div>
`;

exports[`When curve frame has rendered Expect component 2 has rendered successfully 1`] = `
<div
  class="filter_container "
  data-testid="frameCurveTwo"
>
  <div
    class="box_Outer bg-[#FFFFFF]"
    style="padding: 2%;"
  >
    <div
      class="filter_container"
    >
      <div
        class="box_Inner bg-white"
      >
        <span>
          &lt;div}&gt;
                &lt;p&gt;Hello world&lt;/p&gt;
                &lt;p&gt;Hello world&lt;/p&gt;
                &lt;p&gt;Hello world&lt;/p&gt;
                &lt;p&gt;Hello world&lt;/p&gt;
            &lt;/div&gt;
        </span>
      </div>
    </div>
  </div>
  <svg
    height="0"
    style="visibility: hidden; position: absolute;"
    version="1.1"
    width="0"
    xmlns="http://www.w3.org/2000/svg"
  >
    <defs>
      <filter
        id="goo"
      >
        <fegaussianblur
          in="SourceGraphic"
          result="blur"
          stdDeviation="3"
        />
        <fecolormatrix
          in="blur"
          mode="matrix"
          result="goo"
          values="1 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 19 -9"
        />
        <fecomposite
          in="SourceGraphic"
          in2="goo"
          operator="atop"
        />
      </filter>
    </defs>
  </svg>
</div>
`;
